ml-agents/mlagents/trainers/tests/test_simple_rl.py

```diff
@@ -323,3 +330,43 @@ def test_simple_ghost_fails(use_discrete):
     assert any(reward > success_threshold for reward in processed_rewards) and any(
         reward < success_threshold for reward in processed_rewards
     )
+
+
+@pytest.mark.parametrize("use_discrete", [True, False])
+def test_simple_record(use_discrete):
+    env = Record1DEnvironment([BRAIN_NAME], use_discrete=use_discrete, n_demos=100)
+    config = generate_config(PPO_CONFIG)
+    _check_environment_trains(env, config)
+    agent_info_protos = env.demonstration_protos[BRAIN_NAME]
+    meta_data_proto = DemonstrationMetaProto()
+    brain_param_proto = BrainParametersProto(
+        vector_action_size=[1],
+        vector_action_descriptions=[""],
+        vector_action_space_type=discrete if use_discrete else continuous,
+        brain_name=BRAIN_NAME,
+        is_training=True,
+    )
+    action_type = "Discrete" if use_discrete else "Continuous"
+    demo_path = "demos/1DTest" + action_type + ".demo"
+    write_demo(demo_path, meta_data_proto, brain_param_proto, agent_info_protos)
```

Comment:
We should save this to a temporary directory once and use multiple times by the other tests. Pytest provides a way to do this: https://docs.pytest.org/en/latest/tmpdir.html#the-tmpdir-factory-fixture